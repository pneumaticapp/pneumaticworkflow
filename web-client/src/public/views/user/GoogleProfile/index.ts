/* eslint-disable */
/* prettier-ignore */
export * from './GoogleProfile';
