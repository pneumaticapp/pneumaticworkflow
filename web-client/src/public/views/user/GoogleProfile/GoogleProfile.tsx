import * as React from 'react';

import { Avatar } from '../../../components/UI/Avatar';
import { EUserStatus } from '../../../types/user';

import styles from './GoogleProfile.css';

export interface IGoogleProfileProps {
  photo?: string;
  firstName?: string;
  lastName?: string;
  email?: string;
}

export function GoogleProfile({ photo = '', firstName = '', lastName = '', email = '' }: IGoogleProfileProps) {
  if (!email) {
    return null;
  }

  return (
    <div className="d-flex mb-4">
      <div className="d-flex align-items-center mr-3">
        <Avatar
          className={styles['avatar']}
          user={{
            firstName,
            lastName,
            email,
            photo,
            status: EUserStatus.Registration,
          }}
        />
      </div>
      <div className="d-flex flex-column">
        <span>
          {firstName} {lastName}
        </span>
        <span>{email}</span>
      </div>
    </div>
  );
}
