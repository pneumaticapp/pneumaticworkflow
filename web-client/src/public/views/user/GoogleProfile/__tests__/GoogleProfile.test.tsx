/* eslint-disable */
/* prettier-ignore */
import * as React from 'react';
import { shallow } from 'enzyme';

import { GoogleProfile } from '../GoogleProfile';

const mockProps = {
  photo: '/mock/google/photo.jpg',
  firstName: 'Google',
  lastName: 'Profile',
  email: 'google@pneumatic.app',
};

describe('GoogleProfile', () => {
  it('корректно рендерится со всеми данными', () => {
    const wrapper = shallow(<GoogleProfile {...mockProps}/>);

    expect(wrapper).toMatchSnapshot();
  });
  it('если нет email возвращает null', () => {
    const wrapper = shallow(<GoogleProfile {...mockProps} email=""/>);

    expect(wrapper.html()).toBeNull();
  });
});
