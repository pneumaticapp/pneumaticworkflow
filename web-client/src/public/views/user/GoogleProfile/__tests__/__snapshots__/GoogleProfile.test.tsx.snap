// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleProfile корректно рендерится со всеми данными 1`] = `
<div
  className="d-flex mb-4"
>
  <div
    className="d-flex align-items-center mr-3"
  >
    <Avatar
      className="avatar"
      user={
        {
          "email": "google@pneumatic.app",
          "firstName": "Google",
          "lastName": "Profile",
          "photo": "/mock/google/photo.jpg",
          "status": "registration",
        }
      }
    />
  </div>
  <div
    className="d-flex flex-column"
  >
    <span>
      Google
       
      Profile
    </span>
    <span>
      google@pneumatic.app
    </span>
  </div>
</div>
`;
